enterprise/server/util/redis/redis_test.go

```diff
@@ -0,0 +1,309 @@
+package redis_test
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"strconv"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	redisutil "github.com/buildbuddy-io/buildbuddy/enterprise/server/util/redis"
+	redis "github.com/go-redis/redis/v8"
+)
+
+func genRedisOptions(
+	protocol string,
+	user string,
+	password string,
+	hostname string,
+	port string,
```

Comment:
I think if you collapse these into a "hostname:port" string (maybe call it 'interface' or 'addr'?) you might be able to combine this function with the genUnixOptions function below?

Reply:
Did that, as well as an analagous change for the `genURI` functions